crates/subspace-farmer-components/src/plotting.rs

```diff
@@ -198,11 +203,14 @@ async fn plot_pieces_in_batches_non_blocking<PG: PieceGetter>(
         })
         .collect::<FuturesOrdered<_>>();
 
+    let mut pieces = Vec::new();
```

Comment:
This will unfortunately almost double the amount of RAM required during plotting. I see you want to avoid taking `lock()`, but allocation is not free either and I'm not sure RAM usage is justified. We want to process multiple sectors in parallel, with large pieces and large sectors required memory will be significant and saving a few thousand calls to `.lock()` is not very expensive.